art/estimators/certification/interval/pytorch.py

```diff
@@ -0,0 +1,376 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2022
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements certification using interval (box) domain certification.
+
+| Paper link: https://ieeexplore.ieee.org/document/8418593
+"""
+
+from typing import List, Optional, Tuple, Union, Any, TYPE_CHECKING
+
+import logging
+import warnings
+
+import numpy as np
+import torch
+
+from art.estimators.certification.interval.interval import (
+    IntervalConv2D,
+    IntervalDenseLayer,
+    IntervalReLU,
+    IntervalBounds,
+)
```

Comment:
I think we need to update the naming of the tools to avoid future name conflicts when the same methods will be provided in TensorFlow. How about for example `PyTorchIntervalConv2D` instead of `IntervalConv2D`?

Reply:
I agree. For now it's ok, but we could run into problems later on. Best to make the changes now. I'll use the prefix PyTorch as suggested.